examples/bmi-for-nextgen.ipynb

```diff
@@ -0,0 +1,1696 @@
+    "\n",
+    "CSDMS groups functions into six categories:\n",
+    "\n",
+    "1. Model information functions\n",
+    "  - These describe the model and provide ancillary info like the number of input and output variables and their names\n",
+    "  - You can use [CSDMS standard names](https://csdms.colorado.edu/wiki/CSDMS_Standard_Names)\n",
+    "    - e.g. `atmosphere_water__precipitation_leq-volume_flux`\n",
+    "2. Variable information functions\n",
+    "  - These provide variable units, data types, and memory size\n",
+    "  - NextGen uses standard unit definitions from [UDUNITS](https://www.unidata.ucar.edu/software/udunits/) to perform automatic unit conversion\n",
+    "    - e.g. `meters`, `kg m-2`, `mm h-1`, `hours`, `degC`, etc.\n",
+    "3. Time functions\n",
+    "  - These provide the model time step, beginning time, current time, end time\n",
+    "4. Model control functions\n",
+    "  - These start the model, run the model, and finalize the model\n",
+    "5. Variable getter and setter functions\n",
+    "  - These get and set values for implemented model state variables\n",
+    "6. Model grid functions\n",
+    "  - These describe the model's spatial discretization"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b0b4a2b3",
+   "metadata": {},
+   "source": [
+    "# 3 Implementing BMI\n",
+    "\n",
+    "Now that we've seen the wrong way of doing things and had an introduction to BMI, we will walk through the transition from `snowPy` to `snowBMI`. \n",
+    "\n",
+    "## 3.1 Introducting snowBMI\n",
+    "\n",
+    "`snowBMI` is an improved, modularized version of `snowPy` with a BMI implementation available [here on GitHub](https://github.com/SnowHydrology/snowBMI/). The snow accumulation and melt code (i.e., the hydrologic processes) are the same, but almost everything else is different and better.\n",
+    "\n",
+    "For example, while `snowPy` used only a single file without any modularization. `snowBMI` breaks out the code into different files and modules. You can view the former in the directory structure below and we'll cover the latter shortly.\n",
+    "\n",
+    "```\n",
+    "├── requirements.txt\n",
+    "├── setup.cfg\n",
+    "├── setup.py\n",
+    "└── snow\n",
+    "    ├── __init__.py\n",
+    "    ├── _version.py\n",
+    "    ├── bmi_snow.py\n",
+    "    └── snow.py\n",
+    "```\n",
+    "\n",
+    "Although this simplified conversion will likely differ from the more complex steps needed for other models, you can use it as a template or best practice guide in your work.\n",
+    "\n",
+    "## 3.2 Making your model BMI compliant\n",
+    "\n",
+    "In [Section 2.3](#2.3-Making-a-model-BMI-compliant) above, we saw some of the features that make a model BMI compliant. We'll walk through those now.\n",
+    "\n",
+    "### 3.2.1 Compatible language\n",
```

Comment:
I personally hesitate to talk about bmi language compatibility, since that isn't a real thing.  You can define the BMI interface functions and their semantics in ANY language, but implying that models implementing that BMI interface are compatible across languages is dangerous (because it simply is true, BMI makes no language interoperability claims (that I know of??? (and how deep can you nest parenthetical sub comments???))).  BMI Supported lanaguages is a looser definition that I think is more inline with the BMI documentation that doesn't exactly imply cross-langague compatibility or interoperability.

Reply:
Modified to say supported languages